Sources/AppcuesKit/Presentation/Traits/Appcues/AppcuesTargetBehaviorTrait.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  AppcuesTargetBehaviorTrait.swift
+//  AppcuesKit
+//
+//  Created by James Ellis on 4/25/23.
+//  Copyright © 2023 Appcues. All rights reserved.
+//
+
+import UIKit
+
+@available(iOS 13.0, *)
+internal class AppcuesTargetBehaviorTrait: AppcuesBackdropDecoratingTrait {
+    struct Config: Decodable {
+        let actions: [Experience.Action]?
+    }
+
+    enum ActionType: String {
+        case tap
+        case longPress
+    }
+
+    static let type: String = "@appcues/target-behavior"
+
+    weak var metadataDelegate: AppcuesTraitMetadataDelegate?
+
+    private let tapActions: [Experience.Action]
+    private let longPressActions: [Experience.Action]
+
+    private weak var appcues: Appcues?
+
+    private lazy var targetView: UIView = {
+        let view = UIView()
+        view.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(didTapTarget)))
+        view.addGestureRecognizer(UILongPressGestureRecognizer(target: self, action: #selector(didLongPressTarget)))
+        return view
+    }()
+
+    required init?(configuration: AppcuesExperiencePluginConfiguration) {
+        self.appcues = configuration.appcues
+
+        let config = configuration.decode(Config.self)
+        let actions = config?.actions ?? []
+
+        tapActions = actions.filter { ActionType(rawValue: $0.trigger) == .tap }
+        longPressActions = actions.filter { ActionType(rawValue: $0.trigger) == .longPress }
+    }
+
+    func decorate(backdropView: UIView) throws {
+        guard let metadataDelegate = metadataDelegate else {
+            // nothing to do without a targetRectangle from the metadata dictionary
+            targetView.removeFromSuperview()
+            return
+        }
+
+        metadataDelegate.registerHandler(for: Self.type, animating: false) { [weak self] in
+            self?.handle(backdropView: backdropView, metadata: $0)
+        }
+    }
+
+    func undecorate(backdropView: UIView) throws {
+        targetView.removeFromSuperview()
+
+        metadataDelegate?.removeHandler(for: Self.type)
+    }
+
+    private func handle(backdropView: UIView, metadata: AppcuesTraitMetadata) {
+        guard let newTarget: CGRect = metadata["targetRectangle"] else {
+            targetView.frame = .zero
+            targetView.removeFromSuperview()
+            return
+        }
+
+        if targetView.superview != backdropView {
+            targetView.removeFromSuperview()
+            backdropView.addSubview(targetView)
+        }
+
+        targetView.frame = newTarget
+    }
+
+    @objc
+    private func didTapTarget() {
+        guard !tapActions.isEmpty, let actionRegistry = appcues?.container.resolve(ActionRegistry.self) else {
+            return
+        }
+
+        actionRegistry.enqueue(
+            actionModels: tapActions,
+            level: .step,
+            interactionType: "Target Tapped",   // ??
+            viewDescription: "Target Rectangle" // ??
```

Comment:
I'm speculating that including the `appcues:v2:step_interaction` analytics, just like button interactions, is desired here. I'm not totally sure though. Also not sure what the values for interactionType or viewDescription should be in this case, so I put in these initial values to start.

Reply:
moving forward with this step_interaction approach.